### PR TITLE
Add dispersing transistion check

### PR DESCRIPTION
**Pull request by bxue-l2**
This PR adds checks in shared_storage.go to ensure, the state of a blob never transit from confirmed to dispersing

### CODE REVIEW

File: disperser/common/blobstore/shared_storage.go [hunk 1]
```diff
@@ -158,6 +160,17 @@ func (s *SharedBlobStore) MarkBlobConfirmed(ctx context.Context, existingMetadat
 }
 
 func (s *SharedBlobStore) MarkBlobDispersing(ctx context.Context, metadataKey disperser.BlobKey) error {
+	refreshedMetadata, err := s.GetBlobMetadata(ctx, metadataKey)
+	if err != nil {
+		s.logger.Error("error getting blob metadata while marking blobDispersing", "err", err)
+		return err
+	}
+	alreadyConfirmed, _ := refreshedMetadata.IsConfirmed()
```
Comment (jianoaix) [inline disperser/common/blobstore/shared_storage.go:168]: What about just checking if the status is PROCESSING? That's the only valid predecessor for DISPERSING.

Comment (bxue-l2) [reply disperser/common/blobstore/shared_storage.go:170]: updated

File: disperser/common/blobstore/shared_storage.go [hunk 2]
```diff
@@ -158,6 +162,23 @@ func (s *SharedBlobStore) MarkBlobConfirmed(ctx context.Context, existingMetadat
...
+	if alreadyConfirmed {
```
Comment (jianoaix) [inline disperser/common/blobstore/shared_storage.go:171]: This part seems not needed anymore

**Review by jianoaix — APPROVED** [review]

File: disperser/common/blobstore/shared_storage.go [hunk 3]
```diff
@@ -158,6 +162,23 @@ func (s *SharedBlobStore) MarkBlobConfirmed(ctx context.Context, existingMetadat
 }
 
 func (s *SharedBlobStore) MarkBlobDispersing(ctx context.Context, metadataKey disperser.BlobKey) error {
+	refreshedMetadata, err := s.GetBlobMetadata(ctx, metadataKey)
```
Comment (jianoaix) [inline disperser/common/blobstore/shared_storage.go:165]: Have you checked the conditional update which may save one RPC call: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ConditionExpressions.html#Expressions.ConditionExpressions.SimpleComparisons

**Comment by mooselumph** [comment]
Looks good. Should we add a similar check for the `PROCESSING` state as well?

**Pull request merged**